Compilation error

Compilation error with the latest STM32 Arduino core on PlatformIO


Compiling .pio\build\blackpill_f401cc\FrameworkArduino\IPAddress.cpp.o
In file included from .pio\libdeps\blackpill_f401cc\OneWireNg\src/OneWireNg_CurrentPlatform.h:44,
                 from .pio\libdeps\blackpill_f401cc\OneWireNg\src/OneWire.h:17,
                 from .pio\libdeps\blackpill_f401cc\DallasTemperature/DallasTemperature.h:22,
                 from src/Sensors/Ds18b20_Sensors.hpp:3,
                 from src/Components/Ambient_And_Chipset_Sensor.hpp:4,
                 from src\Services/Read_All_Temperature_Sensors.hpp:4,
                 from src\Hp_Proxy_Fan_Controller.hpp:20,
                 from src\Main.cpp:2:
.pio\libdeps\blackpill_f401cc\OneWireNg\src/platform/OneWireNg_ArduinoSTM32.h: In member function 'void OneWireNg_ArduinoSTM32::initDtaGpio(unsigned int, bool)':
.pio\libdeps\blackpill_f401cc\OneWireNg\src/platform/OneWireNg_ArduinoSTM32.h:130:25: error: 'GPIOPort' was not declared in this scope
  130 |         _dtaGpio.gpio = GPIOPort[STM_PORT(_dtaGpio.pinName)];
      |                         ^~~~~~~~
Compiling .pio\build\blackpill_f401cc\FrameworkArduino\Print.cpp.o
Compiling .pio\build\blackpill_f401cc\FrameworkArduino\RingBuffer.cpp.o
In file included from .pio\libdeps\blackpill_f401cc\OneWireNg\src/OneWireNg_CurrentPlatform.h:44,
                 from .pio\libdeps\blackpill_f401cc\OneWireNg\src/OneWire.h:17,
                 from .pio\libdeps\blackpill_f401cc\DallasTemperature\DallasTemperature.h:22,
                 from .pio\libdeps\blackpill_f401cc\DallasTemperature\DallasTemperature.cpp:1:
.pio\libdeps\blackpill_f401cc\OneWireNg\src/platform/OneWireNg_ArduinoSTM32.h: In member function 'void OneWireNg_ArduinoSTM32::initDtaGpio(unsigned int, bool)':
.pio\libdeps\blackpill_f401cc\OneWireNg\src/platform/OneWireNg_ArduinoSTM32.h:130:25: error: 'GPIOPort' was not declared in this scope
  130 |         _dtaGpio.gpio = GPIOPort[STM_PORT(_dtaGpio.pinName)];
      |                         ^~~~~~~~

--- a/src/platform/OneWireNg_ArduinoSTM32.h
+++ b/src/platform/OneWireNg_ArduinoSTM32.h
@@ -18,6 +18,12 @@
 #include "OneWireNg_BitBang.h"
 #include "platform/Platform_TimeCritical.h"
 
+#if STM32_CORE_VERSION >= 0x020c0000
+# define GPIO_PORT GPIOPort_list
+#else
+# define GPIO_PORT GPIOPort
+#endif
+
 /**
  * Arduino STM32 platform GPIO specific implementation.
  */
@@ -127,7 +133,7 @@ protected:
         _dtaGpio.pinName = digitalPinToPinName(pin);
         assert(_dtaGpio.pinName != NC);
 
-        _dtaGpio.gpio = GPIOPort[STM_PORT(_dtaGpio.pinName)];
+        _dtaGpio.gpio = GPIO_PORT[STM_PORT(_dtaGpio.pinName)];
         _dtaGpio.ll_pin = STM_LL_GPIO_PIN(_dtaGpio.pinName);
 
         pinMode(pin, (pullUp ? INPUT_PULLUP : INPUT));

